File: src/main/java/org/umbkosa/kcdevdes/user/UserEntity.java
package org.umbkosa.kcdevdes.user;

import jakarta.persistence.*;
import lombok.*;
import org.umbkosa.kcdevdes.base.BaseEntity;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Getter
@Setter
@Table(name = "users")
@NoArgsConstructor
public class UserEntity extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String username;

    @Column(nullable = false)
    private String email;

    @Column
    private String bio;

    @Enumerated(EnumType.STRING)
    private UserRole role;

    @Enumerated(EnumType.STRING)
    private UserStatus status;

    @Column(nullable = false)
    private String profileImageUrl = "https://example.com/default-profile.png";

    @Column
    private LocalDateTime deletedAt = null;
}

File: src/user/entity/user.entity.ts
import { Token } from '../../auth/entity/token.entity';
import { BaseEntity } from '../../base/base.entity';
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  DeleteDateColumn,
  OneToMany,
} from 'typeorm';
import { USER_ROLE } from './user.role';
import { USER_STATUS } from './user.status';

@Entity()
export class User extends BaseEntity {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ type: 'varchar', length: 255, unique: true })
  username: string;

  @Column({ type: 'varchar', length: 255, unique: true })
  email: string;

  @Column({ type: 'enum', enum: USER_ROLE, default: USER_ROLE.USER })
  role: USER_ROLE;

  @Column({ type: 'text', nullable: true })
  bio?: string;

  @Column({ type: 'timestamp', nullable: true })
  lastLoginAt?: Date;

  @Column({ type: 'enum', enum: USER_STATUS, default: USER_STATUS.ACTIVE })
  status: USER_STATUS;

  @Column({ type: 'text', nullable: true })
  profileImageUrl?: string;

  @DeleteDateColumn({ type: 'timestamp', nullable: true })
  deletedAt?: Date;

  @OneToMany(() => Token, (token) => token.user)
  tokens: Token[];
}
